### PR TITLE
add retry in smoke test

### PR DESCRIPTION
**Pull request by wangzlei**
*Issue #, if available:*
Github action might have random dependency downloading failure. Add retry to reduce false alarmm

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by wangzlei** [comment]
aws-otel-lambda canary failure has been solved by [adding retry](https://github.com/aws-observability/aws-otel-lambda/commit/ce758a9b689b6771df65e25d7e8b8be6ccfb0fba#diff-4b56c7c159ef6d182d11a4f6d189ea69d9ee8b2dfde7e7e920052190549c65e8)
https://github.com/aws-observability/aws-otel-lambda/actions/workflows/canary.yml

File: .github/workflows/continuous-monitoring.yml [hunk 1]
```diff
@@ -32,7 +32,11 @@ jobs:
           
       - name: Run smoke tests
         id: distribution-availability
-        run: ./gradlew :smoke-tests:check -PtestDistributionChannel=true
+        uses: nick-invision/retry@v2
```
Comment (anuraaga) [inline .github/workflows/continuous-monitoring.yml:35]: We should avoid random third party actions if possible, especially in a distribution channel monitor. I think we can update the cache action to cache dependencies as well instead

Comment (wangzlei) [reply .github/workflows/continuous-monitoring.yml:35]: Github action may occur downloading failure in anywhere, not only gradle.zip, if we cannot predict which part would get failure it is hard to add cache, now we just see an error then fix it. Also, adding more cache will inflate CI/CD code.

When we testing if the target binary/dependency is available, say xray-java-sdk in this case, we cannot add cache for it. Retry might be the only way to mitigate false download failure.

File: .github/workflows/continuous-monitoring.yml [hunk 2]
```diff
@@ -32,7 +32,11 @@ jobs:
           
       - name: Run smoke tests
         id: distribution-availability
-        run: ./gradlew :smoke-tests:check -PtestDistributionChannel=true
+        run: |
+          (echo "===== Attempt: 1 ====" && ./gradlew :smoke-tests:check -PtestDistributionChannel=true) || \
+          (echo "===== Attempt: 2 ====" && ./gradlew :smoke-tests:check -PtestDistributionChannel=true) || \
+          (echo "===== Attempt: 3 ====" && ./gradlew :smoke-tests:check -PtestDistributionChannel=true) || \
+          (echo "===== Failed ====" && exit 1)
```
Comment (NathanielRN) [inline .github/workflows/continuous-monitoring.yml:39]:
```suggestion
          (echo "===== No more retries. Failed. ====" && exit 1)
```

**Review by NathanielRN — APPROVED** [review]
Very slick use of shell commands :)

**Review by anuraaga — APPROVED** [review]

**Pull request merged**